docs/onboarding_best_practices.rst

```diff
@@ -0,0 +1,165 @@
+
+- 1 day after offline data becomes available, if push frequency daily 
+- 1 hour after offline data becomes available, if push frequency is hourly
+
+Generally a stream retention of 4-5 days is sufficient, as this is enough time for the offline flow to generate data. The retention of the realtime table should no be more than the stream retention.
+
+4. Pre aggregations
+###################
+Consider reducing the number of stream events per second, by aggregating in a samza job. Keep the aggregation window reasonable (5 minutes, 15 minutes, 1 hour depending on the events per second). If the aggregation window is large (say 6 hours), a burst of events will be sent to Pinot periodically after long gaps of silence. This kind of ingestion pattern is not ideal for Pinot consumption, and a steady stream of events is preferred. 
+
+5. Metrics aggregation in Pinot realtime
+#########################################################
+If aggregation is not possible at source (due to freshness concerns), pinot can aggregate after events are received, by enabling aggregateMetrics: true in table config. Note that Pinot supports only “sum” aggregations in realtime.
+
+6. Aggregation buckets
+#####################################
+Keep the aggregation bucket size the same across realtime and offline. If aggregation buckets are not consistent across realtime and offline, there can be undesirable results for data which spans the boundary. For example, if in realtime you have MILLISECONDS time column rounded off to the nearest hour via the samza job, follow the same model in offline flow, and round off time column to the nearest hour in the pre aggregations. This means, the samza job will produce a new timestamp every hour, whereas the offline segment for the day will have 24 unique timestamps. 
+
+
+
+
+Querying Pinot
+^^^^^^^^^^^^^^
+`PQL doc <https://pinot.readthedocs.io/en/latest/pql_examples.html>`_
+
+1. Apply **time filters** to your queries where applicable, to avoid fetching too much data
+
+2. Apply **reasonable LIMIT and TOP** to your query results, based on what application needs
+
+3. Pinot supports **distinct count**. However it is an expensive operation. Consider using **distinctCountHLL**, which is an approximate distinct count
+
+4. Implement **caching layer on the application side**, to reduce the amount of repeated queries being sent to Pinot
+
+5. For **report generation** style applications, it might be better to **split your query** into smaller time periods
+
+6. At Pinot table side, **hard QPS quota limits** can be applied `tableConfig -> quotaConfig -> maxQueriesPerSecond`. If this quota is exceed, a 429 exception will be thrown with  message 
+
+.. code-block:: none
+
+    "Request <request id> exceeds query quota for table:<table name>, query:<query>"
+
+7. Pinot doesn’t have query throttling. It is recommended to have an **application side query throttling mechanism**
```

Comment:
Combine this with the previous one?
Also, suggested wording: "Pinot does not throttle queries. Applications may choose to throttle queries to minimize 429 returns if necessary, or have administrators provision more hardware to support the rate of queries"